src/mcc_splits.jl

```diff
@@ -141,10 +145,25 @@ end
 #=
 Map split `S[i]` to `t`.
 =#
-function _map_split_to_tree(S::SplitList, i::Integer, t::Tree, treesplits::SplitList)
-	# Not lca in case lca(t, leaves(S,i)) contains extra leaves not in S[i]
-	roots = TreeTools.blca([t.lleaves[x] for x in leaves(S,i)]...)
+function _map_split_to_tree(S::SplitList, i::Integer, t::Tree, treesplits::SplitList; strict=false, MCCs=nothing)
```

Comment:
Yes that is true - I will change that for now. I think it would be good to discuss if we want to use this form of `assign_mccs` as it alters the tree and needs a tree of `TreeTools.MiscData` not `TreeTools.EmptyData`. If we want to keep it like is, I would suggest either adding a convert tree to MiscData type if `resolve` is called with the `strict=true` argument or altering the default TreeNode type. Otherwise I could rewrite the function next week to use dictionaries and not alter the tree.

Reply:
Separate functions added as of commit: https://github.com/PierreBarrat/TreeKnit.jl/pull/20/commits/dc1a3ddfd69e109578992b53a09769fbf1d0a4d9 .

The current plan is to use `strict_resolve!` for creating nwk files at the end of TreeKnit and still use `resolve!` (with masks) during SA and for creating the ARG. We shall continue to convert trees to `MiscData` objects when `strict_resolve!` is used.